### ISSUE TITLE
"Recent Commit Messages" does not show commits that were cancelled

### ISSUE THREAD

**Issue by noamtamim** [ForkIssues/Tracker#720]
## Steps to reproduce
1. Commit something
2. You notice an error in the commit
3. Go to the parent commit, select "Reset <branch> to Here"
4. Fix
5. Commit again.

## Expected
The text I already entered in the commit that was cancelled should be in the "Recent Commit Messages" list.

## Actual
It's not these, and I have to type everything again.

**Comment by DanPristupov** [comment]
Fork shows list of the latest 10 commits in the current branch.

If you remove some commit using reset, then there's no way to request its message anymore.

**Comment by noamtamim** [comment]
@DanPristupov yes, I understand it looks at the log (that obviously doesn't contain the removed commits). But I think it should also keep the history of deleted messages, somewhere, because it's very useful.

**Comment by DanPristupov** [comment]
> it should also keep the history of deleted messages

.. and remove duplicates
.. and keep the sort order when user makes more commits/resets
.. and remember the branch they were created on
.. and save/restore them after restart

This is too complicated.

**Comment by noamtamim** [comment]
I understand the complexity, and it's not a critical feature. But please consider that the usefulness of last X messages (**deleted or not**) is arguably greater than that of last X **existing** messages. Personally, I find that the chance I'd want to use the save comment is higher for a *deleted* commit (because of what I wrote in the post). 
In only [2 weeks of using Fork](https://twitter.com/noamtm/status/1160833835606712320), I've already lost and had to retype 2 commit messages. For now, I have to remember to copy aside before resetting.

With that in mind, another implementation might be keeping a list of last 20 messages, and not taking anything from the log. I understand that unlike just scanning the log you have to do some housekeeping, but it might be worth it.
